scipy/optimize/tests/test_minimize_constrained.py

```diff
@@ -614,6 +614,44 @@ def callback(x, info):
         # Also check existence of the 'niter' attribute, for backward
         # compatibility
         assert_(result.get('niter', -1) == 1)
+    
+    def test_issue_18882(self):
+        # https://github.com/scipy/scipy/issues/18882
+        # Test that success is false if the constraint is not met.
```

Comment:
this test does not actually test that `res.success is False` if the constraints are not met.
The test is also a lot longer than it needs to be.